packages/datadog_flutter_plugin/ios/Classes/SwiftDatadogSdkPlugin.swift

```diff
@@ -8,154 +8,162 @@ import Datadog
-      _ = ddConfigBuilder.set(serviceName: serviceName)
+    internal func initialize(configuration: DatadogFlutterConfiguration) {
+        let ddConfiguration = SwiftDatadogSdkPlugin.buildConfiguration(from: configuration)
+
+        Datadog.initialize(appContext: Datadog.AppContext(),
+                           trackingConsent: configuration.trackingConsent,
+                           configuration: ddConfiguration)
+
+        if let tracingConfiguration = configuration.tracingConfiguration {
+            tracer = DatadogTracesPlugin.instance
+            tracer?.initialize(configuration: tracingConfiguration)
+        }
+        if let rumConfiguration = configuration.rumConfiguration {
+            rum = DatadogRumPlugin.instance
+            rum?.initialize(configuration: rumConfiguration)
+        }
     }
 
-    if let threshold = flutterConfig.additionalConfig["_dd.long_task.threshold"] as? TimeInterval {
-      // `_dd.long_task.threshold` attribute is in milliseconds
-      _ = ddConfigBuilder.trackRUMLongTasks(threshold: threshold / 1_000)
+    internal static func buildConfiguration(from flutterConfig: DatadogFlutterConfiguration) -> Datadog.Configuration {
+        let ddConfigBuilder: Datadog.Configuration.Builder
+        if let rumConfig = flutterConfig.rumConfiguration {
+            ddConfigBuilder = Datadog.Configuration.builderUsing(
+                rumApplicationID: rumConfig.applicationId,
+                clientToken: flutterConfig.clientToken,
+                environment: flutterConfig.env
+            )
+                .set(rumSessionsSamplingRate: rumConfig.sampleRate)
+        } else {
+            ddConfigBuilder = Datadog.Configuration.builderUsing(
+                clientToken: flutterConfig.clientToken,
+                environment: flutterConfig.env
+            )
+        }
+
+        if flutterConfig.nativeCrashReportingEnabled {
+            _ = ddConfigBuilder.enableCrashReporting(using: DDCrashReportingPlugin())
+        }
+
+        if let site = flutterConfig.site {
+            _ = ddConfigBuilder.set(endpoint: site)
+        }
+
+        if let batchSize = flutterConfig.batchSize {
+            _ = ddConfigBuilder.set(batchSize: batchSize)
+        }
+        if let uploadFrequency = flutterConfig.uploadFrequency {
+            _ = ddConfigBuilder.set(uploadFrequency: uploadFrequency)
+        }
+
+        if !flutterConfig.firstPartyHosts.isEmpty {
+            _ = ddConfigBuilder.trackURLSession(firstPartyHosts: Set(flutterConfig.firstPartyHosts))
+        }
+
+        if let customEndpoint = flutterConfig.customEndpoint,
+           let customEndpointUrl = URL(string: customEndpoint) {
+            _ = ddConfigBuilder
+                .set(customLogsEndpoint: customEndpointUrl)
+                .set(customTracesEndpoint: customEndpointUrl)
+                .set(customRUMEndpoint: customEndpointUrl)
+        }
```

Comment:
I see that this is not a change from this PR, but shouldn't we have different URLs on `flutterConfig` as well? one for each product.

Reply:
🤔 It's probably something I'll have to add? At the moment it's only used for integration testing where I haven't needed to set custom URLs per-product. Not sure about how it works with proxies though (and I think we're missing other information to get proxies to work.)